compositor_render/src/transformations/web_renderer/browser.rs

```diff
@@ -0,0 +1,76 @@
+use compositor_chromium::cef;
+use compositor_common::scene::Resolution;
+use crossbeam_channel::{Receiver, Sender};
+
+pub(super) struct BrowserState {
+    painted_frames_receiver: Receiver<Vec<u8>>,
+    frame_data: Vec<u8>,
+}
+
+impl BrowserState {
+    pub fn new(painted_frames_receiver: Receiver<Vec<u8>>) -> Self {
+        Self {
+            painted_frames_receiver,
+            frame_data: Vec::new(),
+        }
+    }
+
+    pub fn retrieve_frame(&mut self) -> &[u8] {
+        if let Some(frame) = self.painted_frames_receiver.try_iter().last() {
+            self.frame_data = frame;
+        }
+
+        &self.frame_data
```

Comment:
```suggestion
        self.frame_data = self.painted_frames_receiver.try_iter().last().or(self.frame_data);
        &self.frame_data
```

Reply:
I don't think it will work. `self.frame_data` is behind mutable reference and `.or()`  takes ownership